### PR TITLE
feat(scripts): optimistic governor scripts

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Optimistic Oracle frontend development requires convenience scripts to generate Optimistic Governor requests.

**Summary**

Adds scripts to deploy Optimistic Governor module, propose/dispute transactions and execute undisputed proposals.

**Details**
- `packages/scripts/src/testnet/OptimisticGovernorDeploy.ts` deploys and initializes Optimistic Governor proxy from provided mastercopy. It is also enabled as a module to the provided Gnosis Safe (module can be enabled only by safe owner if the safe threshold is 1). Alternatively, if no safe address is provided, the script deploys new Gnosis Safe and links it to new Optimistic Governor module.
- `packages/scripts/src/testnet/OptimisticGovernorRequest.ts` proposes token approval transaction through Optimistic Governor that generates Optimistic Oracle V3 assertion. Optionally the script can also dispute the assertion through Optimistic Oracle V3.
- `packages/scripts/src/testnet/OptimisticGovernorExecute.ts` executes undisputed proposal for token approval through Optimistic Governor that also settles Optimistic Oracle V3 assertion after proposal liveness has passed.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [x]  Untested

### CODE REVIEW

File: packages/scripts/src/utils/optimisticGovernorPayload.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  approvalTokenAddress: string;
+  proposalPayload: string;
+  explanation: string;
+  approvalAmount: string;
+  recipient: string;
+}
+
+// Helper function to generate the payload for an approval transaction to be proposed through the OptimisticGovernor.
+// This uses TOKEN, AMOUNT and RECIPIENT from environment variables, or falls back to the provided parameters.
+export async function createApprovalPayload(
```
Comment (mrice32) [inline packages/scripts/src/utils/optimisticGovernorPayload.ts:16]: So this suite of scripts is mainly meant to just test the OG on testnet, right? The approval transaction here is just meant as an example transaction that is hardcoded to give you something to push through, right?

**Review by mrice32 — APPROVED** [review]
Looks good! Just a few minor comments and suggestions!

In general, this PR is very large. In the future, I would suggest breaking this up into smaller PRs (preferably one per script).

File: packages/scripts/src/testnet/OptimisticGovernorDeploy.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      ? utils.parseUnits(process.env.BOND_AMOUNT, decimals)
+      : await mastercopy.bondAmount();
+  const rules = process.env.RULES !== undefined ? process.env.RULES : "proxy rules";
+  const identifier =
+    process.env.IDENTIFIER !== undefined
+      ? utils.formatBytes32String(process.env.IDENTIFIER)
+      : await mastercopy.identifier();
+  const liveness =
+    process.env.LIVENESS !== undefined ? BigNumber.from(process.env.LIVENESS) : await mastercopy.liveness();
+  const saltNonce = Number(new Date());
```
[the remark this answers is not in the document]
Comment (mrice32) [reply packages/scripts/src/testnet/OptimisticGovernorDeploy.ts:78]: And it's not possible for someone to create an exploit by deliberately creating a collision?

File: packages/scripts/src/utils/optimisticGovernorPayload.ts (hunk 1 above)
Comment (mrice32) [reply packages/scripts/src/utils/optimisticGovernorPayload.ts:17]: Gotcha. Makes sense!

File: packages/scripts/src/testnet/OptimisticGovernorDeploy.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Reinis-FRP) [reply packages/scripts/src/testnet/OptimisticGovernorDeploy.ts:78]: Any hash collision could be exploitable only if gnosis safe signers enabled expected OG module address before OG contract was deployed. But this is not the expected usage flow and would require the "attacked" user to deliberately modify this scrip to reverse the order when module is enabled. And the attacker would not be able to practically generate hash collision with different initialization parameters within such short time frame. The only feasible attack would be to frontrun the user in deploying the module with the same salt/initialization parameters, but it should not be a concern as such module would retain the same usability as required by "attacked" user.

**Pull request merged**